app/api/items/items.go

```diff
@@ -59,3 +60,87 @@ func checkHintOrScoreTokenRequiredFields(user *database.User, taskToken *token.T
 	}
 	return service.NoError
 }
+
+type permission struct {
+	ItemID                     int64
+	CanViewGeneratedValue      int
+	CanGrantViewGeneratedValue int
+	CanWatchGeneratedValue     int
+	CanEditGeneratedValue      int
+}
+
+type itemChild struct {
+	// required: true
+	ItemID int64 `json:"item_id,string" sql:"column:child_item_id" validate:"set,child_item_id"`
+	// default: 0
+	Order int32 `json:"order" sql:"column:child_order"`
+}
+
+type insertItemItemsSpec struct {
+	ParentItemID               int64
+	ChildItemID                int64
+	Order                      int32
+	ContentViewPropagation     string
+	UpperViewLevelsPropagation string
+	GrantViewPropagation       bool
+	WatchPropagation           bool
+	EditPropagation            bool
+}
+
+func constructItemsItemsForChildren(childrenPermissions []permission, children []itemChild,
```

Comment:
Even if private, most non-trivial function should a small spec / comment describing what it does.

Reply:
added comments

Reply:
👍 
Try, by default, to add a bit more comments in the code (not especially code spec but a few words of explanation for a non-trivial block of code).